add_js_file(..., async="async") causes SyntaxError

**Is your feature request related to a problem? Please describe.**
I want to put `<script src="..." async="async"></script>` tag in built HTML.
I found the example in https://www.sphinx-doc.org/en/master/extdev/appapi.html#sphinx.application.Sphinx.add_js_file and I tried it.

In conf.py,

```python
def setup(app):
    app.add_js_file('example.js', async="async")
```

```shell
$ make html
Sphinx v4.3.1 を実行中

Configuration error:
設定ファイルに文法エラーが見つかりました: invalid syntax (conf.py, line 63)

make: *** [html] Error 2
```

Environment

- Python 3.9.4
- Sphinx 4.3.1

*Problem*: We cannot use `async` as parameter of `add_js_file` in Python 3.7 or later.
https://docs.python.org/3/whatsnew/3.7.html#summary-release-highlights
>async and await are now reserved keywords.

**Describe the solution you'd like**
Proposal: add a parameter to `add_js_file` to specify async or not.
`add_js_file(filename: str, priority: int = 500, load_async: bool = False, **kwargs: Any)`

```
app.add_js_file('example.js', load_async=False)
# => <script src="_static/example.js"></script>

app.add_js_file('example.js', load_async=True)
# => <script src="_static/example.js" async></script>
```

**Describe alternatives you've considered**
If my proposal is not implemented, I would like to update the example of the document https://www.sphinx-doc.org/en/master/extdev/appapi.html#sphinx.application.Sphinx.add_js_file.

Options:

1. Show Python version limitation

```
app.add_js_file('example.js', async="async")  # only Python 3.6 or earlier (cause SyntaxError in Python 3.7 or later)
# => <script src="_static/example.js" async="async"></script>
```

2. Remove `async="async"` example
In 2022, Python 3.6 will no longer be supported.
ref: https://www.python.org/dev/peps/pep-0494/#lifespan
The example causes `SyntaxError` in Python 3.7 and later, so It seems better to remove it than to keep it in the document.

**Additional context**
N/A

--- a/sphinx/application.py
+++ b/sphinx/application.py
@@ -933,14 +933,18 @@ class Sphinx:
     def add_js_file(self, filename: str, priority: int = 500, **kwargs: Any) -> None:
         """Register a JavaScript file to include in the HTML output.
 
-        Add *filename* to the list of JavaScript files that the default HTML
-        template will include in order of *priority* (ascending).  The filename
-        must be relative to the HTML static path , or a full URI with scheme.
-        If the priority of the JavaScript file is the same as others, the JavaScript
-        files will be included in order of registration.  If the keyword
-        argument ``body`` is given, its value will be added between the
-        ``<script>`` tags. Extra keyword arguments are included as attributes of
-        the ``<script>`` tag.
+        :param filename: The filename of the JavaScript file.  It must be relative to the HTML
+                         static path, a full URI with scheme, or ``None`` value.  The ``None``
+                         value is used to create inline ``<script>`` tag.  See the description
+                         of *kwargs* below.
+        :param priority: The priority to determine the order of ``<script>`` tag for
+                         JavaScript files.  See list of "prority range for JavaScript
+                         files" below.  If the priority of the JavaScript files it the same
+                         as others, the JavaScript files will be loaded in order of
+                         registration.
+        :param kwargs: Extra keyword arguments are included as attributes of the ``<script>``
+                       tag.  A special keyword argument ``body`` is given, its value will be
+                       added between the ``<script>`` tag.
 
         Example::
 
@@ -984,12 +988,14 @@ class Sphinx:
     def add_css_file(self, filename: str, priority: int = 500, **kwargs: Any) -> None:
         """Register a stylesheet to include in the HTML output.
 
-        Add *filename* to the list of CSS files that the default HTML template
-        will include in order of *priority* (ascending).  The filename must be
-        relative to the HTML static path, or a full URI with scheme.  If the
-        priority of the CSS file is the same as others, the CSS files will be
-        included in order of registration.  The keyword arguments are also
-        accepted for attributes of ``<link>`` tag.
+        :param filename: The filename of the CSS file.  It must be relative to the HTML
+                         static path, or a full URI with scheme.
+        :param priority: The priority to determine the order of ``<link>`` tag for the
+                         CSS files.  See list of "prority range for CSS files" below.
+                         If the priority of the CSS files it the same as others, the
+                         CSS files will be loaded in order of registration.
+        :param kwargs: Extra keyword arguments are included as attributes of the ``<link>``
+                       tag.
 
         Example::
 

